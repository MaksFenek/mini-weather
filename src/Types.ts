export interface IWeatherData {
  weather: [
    {
      main: string;
      description: string;
    }
  ];
  main: {
    temp: number;
    pressure: number;
    humidity: number;
  };
  wind: {
    speed: number;
    deg: number;
  };
  coord: {
    lon: number;
    lat: number;
  };
}

export interface IDay {
  name: string;
  temp: {
    day: number;
    night: number;
  };
  pressure: number;
  humidity: number;
  weather: [
    {
      main: string;
    }
  ];
  wind_speed: number;
  wind_deg: number;
}

export interface IForecastData {
  daily: IDay[];
}
