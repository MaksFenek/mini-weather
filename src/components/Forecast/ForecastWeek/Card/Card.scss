.card {
  background: #bbdefb;
  border-radius: 20px;
  min-width: 150px;
  width: 100%;
  margin: 0 2px;
  display: flex;
  flex-direction: column;
  align-items: center;

  &-day {
    font-size: 20px;
    font-weight: bold;
    margin-top: 12px;
    margin-bottom: 6px;
  }

  &-icon {
    margin-bottom: 12px;
  }

  &-weather {
    span {
      font-weight: bold;
      font-size: 20px;
    }
  }

  &-params {
    padding: 0;
    margin: 0;
    margin-top: 12px;
    li {
      display: flex;
      align-items: center;
      list-style: none;

      span {
        font-weight: bold;
        margin-left: 14px;
      }
    }
  }
}
