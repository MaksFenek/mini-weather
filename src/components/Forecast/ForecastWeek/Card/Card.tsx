import React from 'react';
import './Card.scss';
import { IDay } from '../../../../Types';

const Card: React.FC<IDay> = ({
  temp,
  pressure,
  humidity,
  weather,
  wind_deg,
  wind_speed,
  name,
}) => {
  return (
    <div className='card'>
      <div className='card-day' aria-label='name'>
        {name}
      </div>
      <svg className='card-icon' height='84px' width='84px'>
        <use
          aria-label='icon'
          xlinkHref={`icons/icons.svg#${weather[0].main.toLowerCase()}`}
        />
      </svg>
      <div className='card-weather'>
        <span aria-label='temp-day'>{temp.day}</span> °C /{' '}
        <span aria-label='temp-night'>{temp.night}</span> °C
      </div>

      <ul className='card-params'>
        <li>
          <svg className='icon' height='36px' width='36px'>
            <use xlinkHref={`icons/params.svg#humidity`} />
          </svg>
          <p>
            <span aria-label='humidity'>{humidity}</span> %
          </p>
        </li>
        <li>
          <svg className='icon' height='36px' width='36px'>
            <use xlinkHref={`icons/params.svg#barometer`} />
          </svg>
          <p>
            <span aria-label='pressure'>{pressure}</span> hPa
          </p>
        </li>
        <li>
          <svg
            className='icon'
            height='52px'
            width='52px'
            style={{ margin: '-7px' }}
          >
            <use
              xlinkHref={`icons/compass.svg#main
              `}
            ></use>
            <use
              style={{
                transform: `rotate(${wind_deg + 270}deg)`,
                transformOrigin: 'center',
              }}
              xlinkHref={`icons/compass.svg#arrow`}
            ></use>
          </svg>
          <p>
            <span aria-label='wind_speed'>
              {wind_speed ? wind_speed : '--'}
            </span>{' '}
            mph
          </p>
        </li>
      </ul>
    </div>
  );
};

export default Card;
