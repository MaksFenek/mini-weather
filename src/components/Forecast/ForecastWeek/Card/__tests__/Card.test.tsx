import React from 'react';
import Card from '../Card';
import { mount } from 'enzyme';
import { IDay } from '../../../../../Types';
import renderer from 'react-test-renderer';

describe('<Card/>', () => {
  const props: IDay = {
    temp: {
      day: 23,
      night: 20,
    },
    name: 'Tuesday',
    humidity: 23,
    pressure: 1000,
    weather: [
      {
        main: 'clouds',
      },
    ],
    wind_deg: 23,
    wind_speed: 2.3,
  };

  const wrapper = mount(<Card {...props} />);
  const tree = renderer.create(<Card {...props} />).toJSON();

  it('should render correctly', () => {
    expect(wrapper.find(`[aria-label='name']`).text()).toEqual(props.name);
    expect(wrapper.find(`[aria-label='temp-day']`).text()).toEqual(
      `${props.temp.day}`
    );
    expect(wrapper.find(`[aria-label='temp-night']`).text()).toEqual(
      `${props.temp.night}`
    );
    expect(wrapper.find(`[aria-label='humidity']`).text()).toEqual(
      `${props.humidity}`
    );
    expect(wrapper.find(`[aria-label='pressure']`).text()).toEqual(
      `${props.pressure}`
    );
    expect(wrapper.find(`[aria-label='wind_speed']`).text()).toEqual(
      `${props.wind_speed}`
    );
    expect(tree).toMatchSnapshot();
  });
});
