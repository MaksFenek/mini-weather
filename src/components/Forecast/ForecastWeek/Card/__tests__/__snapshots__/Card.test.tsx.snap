// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should render correctly 1`] = `
<div
  className="card"
>
  <div
    aria-label="name"
    className="card-day"
  >
    Tuesday
  </div>
  <svg
    className="card-icon"
    height="84px"
    width="84px"
  >
    <use
      aria-label="icon"
      xlinkHref="icons/icons.svg#clouds"
    />
  </svg>
  <div
    className="card-weather"
  >
    <span
      aria-label="temp-day"
    >
      23
    </span>
     °C /
     
    <span
      aria-label="temp-night"
    >
      20
    </span>
     °C
  </div>
  <ul
    className="card-params"
  >
    <li>
      <svg
        className="icon"
        height="36px"
        width="36px"
      >
        <use
          xlinkHref="icons/params.svg#humidity"
        />
      </svg>
      <p>
        <span
          aria-label="humidity"
        >
          23
        </span>
         %
      </p>
    </li>
    <li>
      <svg
        className="icon"
        height="36px"
        width="36px"
      >
        <use
          xlinkHref="icons/params.svg#barometer"
        />
      </svg>
      <p>
        <span
          aria-label="pressure"
        >
          1000
        </span>
         hPa
      </p>
    </li>
    <li>
      <svg
        className="icon"
        height="52px"
        style={
          Object {
            "margin": "-7px",
          }
        }
        width="52px"
      >
        <use
          xlinkHref="icons/compass.svg#main
              "
        />
        <use
          style={
            Object {
              "transform": "rotate(293deg)",
              "transformOrigin": "center",
            }
          }
          xlinkHref="icons/compass.svg#arrow"
        />
      </svg>
      <p>
        <span
          aria-label="wind_speed"
        >
          2.3
        </span>
         
        mph
      </p>
    </li>
  </ul>
</div>
`;
